plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4152,9 +4169,34 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
                 attribute.echoError("Location is not a valid Sign block.");
                 return null;
             }
-            return new ElementTag(((Sign) state).getColor());
+            if (NMSHandler.getVersion().isAtMost(NMSVersion.v1_19)) {
+                return new ElementTag(((Sign) state).getColor());
+            }
+            String side = attribute.hasParam() ? attribute.getParam() : "front";
+            return new ElementTag(((Sign) state).getSide(Side.valueOf(side.toUpperCase())).getColor());
         });
 
+        // <--[tag]
+        // @attribute <LocationTag.sign_waxed>
+        // @returns ElementTag(Boolean)
+        // @mechanism LocationTag.sign_waxed
+        // @group world
+        // @description
+        // Returns whether the location is a Sign block that is waxed.
+        // See also <@link tag LocationTag.sign_glowing>
+        // -->
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
+            tagProcessor.registerTag(ElementTag.class, "sign_waxed", (attribute, object) -> {
+                BlockState state = object.getBlockStateForTag(attribute);
+                if (!(state instanceof Sign)) {
```

Comment:
Should use `instanceof` pattern matching, that way you can use `object.getBlockStateForTag(attribute)` directly in the `instanceof` check instead of storing it separately and casting later